Hospital.API/Services/Resources/EquipmentService.cs

```diff
@@ -115,5 +122,39 @@ public IEnumerable<EquipmentDto> GetEquipmentWithQuantityByType(string equipment
             }
             return allEquipment.Values.ToList();
         }
+
+        public bool RelocateEquipment(EquipmentRelocationDto eqRealDto)
+        {
+            List<EquipmentUnit> equipmentsInRoom = GetEquipmentByRoomIdAndType(eqRealDto.SourceRoomId, eqRealDto.EquipmentType).ToList();
+            if (CheckAmount(eqRealDto.Amount, eqRealDto.SourceRoomId, eqRealDto.EquipmentType))
+            {
+                MoveEquipmentUnits(eqRealDto, equipmentsInRoom);
+                return true;
+            }
+            return false;
+        }
+
+        private void MoveEquipmentUnits(EquipmentRelocationDto eqRealDto, List<EquipmentUnit> equipmentsInRoom)
+        {
+            for (int i = 0; i < eqRealDto.Amount; i++)
+            {
+                equipmentsInRoom[i].CurrentLocation = null;
+                equipmentsInRoom[i].CurrentLocationId = eqRealDto.DestinationRoomId;
```

Comment:
You need to set CurrentLocation here I think, don't put null.

Reply:
We must put null, because CurrentLocation can't have any value before id is set. We tried already, but if it is not null, it can't be changed.

Reply:
Then change after you set id.

Reply:
Then it will stay null.